mapviz_plugins/src/gps_plugin.cpp

```diff
@@ -185,19 +126,23 @@ namespace mapviz_plugins
 
     StampedPoint stamped_point;
     stamped_point.stamp = gps->header.stamp;
-
+    stamped_point.source_frame = local_xy_util_.Frame();
     double x;
     double y;
     local_xy_util_.ToLocalXy(gps->latitude, gps->longitude, x, y);
 
     stamped_point.point = tf::Point(x, y, gps->altitude);
-
-    // The GPS "track" is in degrees, but createQuaternionFromYaw expects radians.
+    lap_checked_ = ui_.show_laps->isChecked();
+    // The GPS "track" is in degrees, but createQuaternionFromYaw expects
+    // radians.
     // Furthermore, the track rotates in the opposite direction and is also
     // offset by 90 degrees, so all of that has to be compensated for.
-    stamped_point.orientation = tf::createQuaternionFromYaw((-gps->track * (M_PI / 180.0)) + M_PI_2);
+    stamped_point.orientation =
+        tf::createQuaternionFromYaw((-gps->track * (M_PI / 180.0)) + M_PI_2);
 
-    if (points_.empty() || stamped_point.point.distance(points_.back().point) >= position_tolerance_)
+    if (points_.empty() ||
+        stamped_point.point.distance(points_.back().point) >=
+            position_tolerance_)
```

Comment:
In compound conditionals like this, especially wrapped over multiple lines, adding some clarifying parentheses would help:
```
if (points_.empty() ||
    (stamped_point.point.distance(points_.back().point) >= position_tolerance_))
```